docs/DeveloperGuide.md

```diff
@@ -1,34 +1,561 @@
+
+As seen from the sequence diagram above, this is the flow of a Grade command.
+The GradeCommandParser parses the user's input, and assigns the relevant attributes in the Grade Command constructor, such as `moduleCredit` and `grade`
+When Grade Command executes,a CommandResult object is created that shows the user the result of the command through TextUi, using `showOutputToUser()`
+
+Given below is an example usage scenario and how the grade feature behaves at each step.
+
+1. The user launches the application. The user inputs `add -m CS2101` into ra.VI, as the user wants to note down 
+a module named ‘CS2101’ and add it to their module list. This input is received by the Ui ,which processes it into 
+a string. The parser parses the string and allocates it to the AddCommand where it is added to the list of modules. 
+
+2. The user inputs `grade CS2101 4 A+`. The parser parses and allocates the user input to GradeCommand. 
+`GradeCommand#execute()` is called and moduleManager checks if such a module exists in the user’s module list, 
+then checks if the input grade is valid according to the NUS grading schematic and finally assigns the specific module, 
+the grade and module credits.
+
+3. The `CommandResult` returns the success message to show the user that their module has successfully been graded. 
+Otherwise, an exception message will be shown regarding the exception caught.
+
+### Timetable Feature
+This feature is facilitated by the TimeTableManager class and TimeTableCommand class.
+Extending from the abstract TimeTableCommand class are the TimeTableAddCommand, TimeTableDeleteCommand and TimeTableViewCommand classes.
+* AddLesson - Add a Lesson to the timetable through `TimeTableManager.addLesson()`
+* DeleteLesson - Delete all associated Lessons from the timetable through `TaskManager.deleteLesson()`
+* ViewTimeTable - List all Lessons in the timetable through `TaskManager.getSpecificDayLessons()` or `TaskManager.getSpecifiedWeekLessons()`
+
+#### Add lesson/s to timetable
+Given below is an example scenario to add a lesson to the timetable and how the timetable feature behaves at each step.
+
+1. The user launches the application for the first time. ra.VI asks for the current NUS week. This input is parsed and 
+initialises the TimeTableManager. 
+
+2. The user inputs `add -m CS2101`, as the user wants to note down a module named `CS2101` and add it to their module list.
+
+3. The user inputs `timetable -add CS2101 TUESDAY 0800 1000 LECTURE 1`. This means the user wants to add a `CS2101 LECTURE` 
+that occurs once a week on `TUESDAY 0800 1000`. This command will be parsed and eventually returns a TimeTableAddCommand.
+
+Step 4. The TimeTableAddCommand is executed, returning a `CommandResult` containing a success message if the Lessons have 
+been successfully added. Otherwise, an exception message will be shown explaining the exception to the user.\
+Common reasons for failure include:
```

Comment:
The earlier 3 points do not use the "Step" prefix, should this be amended as well?